### ISSUE TITLE
Defining Global Variables Across JS, Pug, and Stylus Not Working

### ISSUE THREAD

**Issue by michaelalhilly** [vuejs/vue-cli#5860]
### Version
4.5.4

### Reproduction link
[https://github.com/michaelalhilly/vuecliglobals](https://github.com/michaelalhilly/vuecliglobals)

### Environment info
```
  System:
    OS: Linux 4.19 Debian GNU/Linux 9 (stretch) 9 (stretch)
    CPU: (6) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  Binaries:
    Node: 12.18.3 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.6 - /usr/local/bin/npm
  Browsers:
    Chrome: Not Found
    Firefox: Not Found
  npmPackages:
    @ant-design-vue/babel-helper-vue-transform-on:  1.0.1 
    @vue/babel-helper-vue-jsx-merge-props:  1.0.0 
    @vue/babel-plugin-transform-vue-jsx:  1.1.2 
    @vue/babel-preset-app:  4.5.4 
    @vue/babel-preset-jsx:  1.1.2 
    @vue/babel-sugar-functional-vue:  1.1.2 
    @vue/babel-sugar-inject-h:  1.1.2 
    @vue/babel-sugar-v-model:  1.1.2 
    @vue/babel-sugar-v-on:  1.1.2 
    @vue/cli-overlay:  4.5.4 
    @vue/cli-plugin-babel: ~4.5.0 => 4.5.4 
    @vue/cli-plugin-eslint: ~4.5.0 => 4.5.4 
    @vue/cli-plugin-pwa: ~4.5.0 => 4.5.4 
    @vue/cli-plugin-router: ~4.5.0 => 4.5.4 
    @vue/cli-plugin-vuex: ~4.5.0 => 4.5.4 
    @vue/cli-service: ~4.5.0 => 4.5.4 
    @vue/cli-shared-utils:  4.5.4 
    @vue/component-compiler-utils:  3.2.0 
    @vue/preload-webpack-plugin:  1.1.2 
    @vue/web-component-wrapper:  1.2.0 
    eslint-plugin-vue: ^6.2.2 => 6.2.2 
    vue: ^2.6.11 => 2.6.12 
    vue-cli-plugin-pug: ~1.0.7 => 1.0.7 
    vue-eslint-parser:  7.1.0 
    vue-hot-reload-api:  2.3.4 
    vue-loader:  15.9.3 (16.0.0-beta.5)
    vue-router: ^3.2.0 => 3.4.3 
    vue-style-loader:  4.1.2 
    vue-template-compiler: ^2.6.11 => 2.6.12 
    vue-template-es2015-compiler:  1.9.1 
    vuex: ^3.4.0 => 3.5.1 
  npmGlobalPackages:
    @vue/cli: 4.5.4
```

### Steps to reproduce
npm run serve

### What is expected?
Global variable pop defined in vue.config.js should be available in js, pug, and stylus files as such:

pop.colors.primary

### What is actually happening?
Global variable pop is not defined in any of the files.

---
I'm able to pass this variable to process.env which is then only available in js files as: process.env.pop, but this should not be an issue and these loaders are all capable of receiving global variables.

Is this a limitation with Vue CLI? If not the docs should show users how to set this up.

**Comment by sodatea** [comment]
I'm sorry but the linked repository does not exist.

**Comment by sodatea** [comment]
Closing due to inactivity

**Comment by michaelalhilly** [comment]
@sodatea Apologies. Yes I deleted the repo and should have closed this.

I managed to achieve this by passing the global object to each loader individually, 10 loaders to be exact. Additionally, support for Stylus seems to be diminishing and is much better for Sass so I switched to simplify. It seems Vue-cli config does limit the passing of objects to process.env for js files which is fine. I also had to create a sass function to expose the global object to sass/scss files.

This was all very tedious and it would be nice if Vue-cli exposed a property (i.e. global) which we can use to expose global data to every file compiled. However, if no one else is asking for this then I guess it's irrelevant.

Thanks for your time on this.
